File: rb/features.rb
# WaifuPics SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WaifuPicsFeatures
  def self.make_feature(name)
    case name
    when "base"
      WaifuPicsBaseFeature.new
    when "ratelimit"
      WaifuPicsRatelimitFeature.new
    when "retry"
      WaifuPicsRetryFeature.new
    when "test"
      WaifuPicsTestFeature.new
    when "timeout"
      WaifuPicsTimeoutFeature.new
    else
      WaifuPicsBaseFeature.new
    end
  end
end

File: py/waifupics_sdk/features.py
# WaifuPics SDK feature factory

from waifupics_sdk.feature.base_feature import WaifuPicsBaseFeature
from waifupics_sdk.feature.test_feature import WaifuPicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WaifuPicsBaseFeature(),
        "test": lambda: WaifuPicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
